### PR TITLE
JSON API: test for ticket 6351 with pruning

### PR DESCRIPTION
**Pull request by raphael-speyer-da**
For the latest developments in https://digitalasset.atlassian.net/browse/ETX-421

Concretely, they're running a test with the following setup.

1. A ledger exists, which contains a contract created by Alice but visible by Bob
2. The ledger is pruned
3. A JSON API instance is started with `start-mode=create-and-start`, i.e. from a fresh cache
4. Alice queries JSON API for this contract, and the cache is populated with this contract and her `last_offset` is set
5. Bob queries JSON API for this contract. He has no `last_offset`, so with the logic from https://github.com/digital-asset/daml/pull/18636 we now attempt to populate the cache for Bob using transactions (from `LedgerBegin`) rather than using the ACS. The intent here is to detect the case where Alice may have archived the contract that was already populated.

Because the ledger has been pruned, attempting to get transactions since `LedgerBegin` will fail with `PARTICIPANT_PRUNED_DATA_ACCESSED`

This problem will not occur in cases where the operator has a process of rewarming the cache before every prune - which is a prudent thing to do anyway, to avoid performance surprises. However the test setup above does not conform to that constraint, and if the cache ever needed to be wiped and rebuilt in a production environment, the above problem could still occur.

This PR is simply trying to reproduce this failure. The fix is a follow-up PR.

The sequence of commits and their CI results gives us information:
* [1afea10](https://github.com/digital-asset/daml/pull/18878/commits/1afea107a85de27d9bbe6d2e092788057a30b22d) just enables building and running tests for oracle with canton-ee
* [b7ce6c0](https://github.com/digital-asset/daml/pull/18878/commits/b7ce6c05942a323e48f05ef9953cc35e88468dbe) adds the new test, which fails in the expected way. The relevant tests can be run with 
  `bazel test --config=oracle --test_tag_filters=canton-ee //ledger-service/...`
* [e26b51a](https://github.com/digital-asset/daml/pull/18878/commits/e26b51aad76e08805e155cd4500626277a7ef12f) `ignore`s the test again. It will be un-ignored in the fix PR. This goes green in CI.
* [10cffad](https://github.com/digital-asset/daml/pull/18878/commits/10cffadcf69c24de65bf0ed2a09d1c04469ecbb3)  reverts `1afea10`

### CODE REVIEW

File: sdk/ledger-service/http-json/src/it/edition/ee/com/daml/http/PruningTest.scala
Comment (raphael-speyer-da) [inline sdk/ledger-service/http-json/src/it/edition/ee/com/daml/http/PruningTest.scala:1]: This test was moved into
`sdk/ledger-service/http-json/src/itlib/scala/http/AbstractPruningTest.scala`

File: sdk/ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala [hunk 1]
```diff
@@ -368,55 +368,6 @@ abstract class QueryStoreAndAuthDependentIntegrationTest
       }
     }
 
-    "query for visible contract, which was created and archived by another party before your first query" in withHttpService {
```
Comment (raphael-speyer-da) [inline sdk/ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala:371]: This test was also moved into
`sdk/ledger-service/http-json/src/itlib/scala/http/AbstractPruningTest.scala`
which has both Postgres and Oracle concrete classes.

The difference between this version and the new one is that the new one also adds a prune step at the beginning, which is the situation we're trying to test for.

File: sdk/ledger-service/http-json/src/itlib/scala/http/AbstractPruningTest.scala [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        (alice, aliceHeaders) <- fixture.getUniquePartyAndAuthHeaders("Alice")
+        (bob, bobHeaders) <- fixture.getUniquePartyAndAuthHeaders("Bob")
+
+        // perform some actions on the ledger
+        (_, offsetAfterArchive) <- offsetBeforeAfterArchival(
+          alice,
+          fixture,
+          aliceHeaders,
+        )
+        // prune
```
Comment (raphael-speyer-da) [inline sdk/ledger-service/http-json/src/itlib/scala/http/AbstractPruningTest.scala:69]: This is what we're adding to the pre-existing test that we moved out of 
`sdk/ledger-service/http-json/src/itlib/scala/http/AbstractHttpServiceIntegrationTest.scala`

**Review by erwin-ramirez-da — APPROVED** [review]
good that test good moved to relative place.. LGTM

**Pull request merged**